Sources/Datadog/Core/Persistence/Reading/FileReader.swift

```diff
@@ -47,52 +47,66 @@ internal final class FileReader: Reader {
         }
     }
 
-    /// Decrypts data if encryption is available.
-    ///
-    /// When encryption is provided, the data is splitted using data-format separator, each slices
-    /// is then decoded from base64 and decrypted. Data is finally re-joined with data-format separator.
+    /// Decodes input data
     ///
-    /// If no encryption, the data is returned.
+    /// The input data is expected to be a stream of `DataBlock`. Only block of type `event` are
+    /// consumed and decrypted if encryptio is available. Decrypted events data are finally joined with
+    /// data-format separator.
     ///
-    /// - Parameter data: The data to decrypt.
-    /// - Returns: Decrypted data.
-    private func decrypt(data: Data) -> Data {
-        guard let encryption = encryption else {
-            return data
-        }
+    /// - Parameter data: The data to decode.
+    /// - Returns: The decoded and formatted data.
+    private func decode(data: Data) throws -> Data {
+        let reader = DataBlockReader(data: data)
 
         var failure: String? = nil
         defer {
             failure.map { userLogger.error($0) }
         }
 
-        return data
-            // split data
-            .split(separator: dataFormat.separatorByte)
-            // decode base64 - report failure
+        // get event blocks only
+        return try reader.all()
             .compactMap {
-                if let data = Data(base64Encoded: $0) {
-                    return data
+                switch $0.type {
+                case .event:
+                    return $0.data
+                default:
+                    return nil
```

Comment:
Better to have explicit `case` list - it's a core piece of our logic, so let's leverage type safety.

Reply:
I added another case `BlockType` as an example, it will contain only `event` and the `default` here will be gone :)